packages/fxa-settings/src/models/contexts/AppContext.ts

```diff
@@ -24,13 +28,94 @@ export interface AppContextValue {
   config?: Config;
   account?: Account;
   session?: Session; // used exclusively for test mocking
+  uniqueUserId?: string; // used for experiments
+  experiments?: any; // TODO: add types for experiments
 }
 
 export interface SettingsContextValue {
   alertBarInfo?: AlertBarInfo;
   navigatorLanguages?: readonly string[];
 }
 
+/**
+ * Fetches or generates a new client ID that is stable for that browser client/cookie jar.
+ *
+ * N.B: Implemenation is taken from `fxa-content-server/.../models/unique-user-id.js` with
+ * inlined code that was written using Backbone utilities which could not immediately be transferred over.
+ * @returns a new or existing UUIDv4 for this user.
+ */
+function getUniqueUserId(): string {
+  function resumeTokenFromSearchParams(): string | null {
+    // Check the url for a resume token, that might have a uniqueUserId
+    const searchParams = new URLSearchParams(window.location.search);
+    const resumeToken = searchParams.get('resume');
+    return resumeToken;
+  }
+
+  function maybePersistFromToken(resumeToken: string) {
+    // populateFromStringifiedResumeToken - fxa-content-server/.../mixins/resume-token.js
+    if (resumeToken) {
+      try {
+        // createFromStringifiedResumeToken - fxa-content-server/.../models/resume-token.js
+        const resumeTokenObj = JSON.parse(atob(resumeToken));
+        if (typeof resumeTokenObj?.uniqueUserId === 'string') {
+          // Key name is derived from the Local Storage implementation.
+          // fullKey - fxa-content-server/.../lib/storage.ts
+          localStorage.setItem(
+            `__fxa_storage.uniqueUserId`,
+            resumeTokenObj?.uniqueUserId
+          );
+          // Use uuid provided by resume token
+          return resumeTokenObj?.uniqueUserId;
+        }
+      } catch (error) {
+        Sentry.captureMessage('Failed parse resume token.', {
+          extra: {
+            resumeToken: resumeToken.substring(0, 10) + '...',
+          },
+        });
+      }
+    }
+  }
+
+  function fetch(): string | null {
+    return localStorage.getItem('__fxa_storage.uniqueUserId');
+  }
+
+  function persist(uniqueUserId: string) {
+    localStorage.setItem(`__fxa_storage.uniqueUserId`, uniqueUserId);
+  }
+
+  const resumeToken = resumeTokenFromSearchParams();
+  if (resumeToken) {
+    maybePersistFromToken(resumeToken);
+  }
+
+  // Check local storage for an existing resume token
+  let uniqueUserId = fetch();
+
+  // Generate a new token if one is not found!
+  if (!uniqueUserId) {
+    uniqueUserId = uuid();
+    persist(uniqueUserId);
+  }
+
+  return uniqueUserId;
+}
+
+function fetchNimbusExperiments(uniqueUserId: string) {
+  // We reuse parseAcceptLanguage with navigator.languages because
+  // that is the same as getting the headers directly as stated on MDN.
+  // See: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages
+  const [locale] = parseAcceptLanguage(navigator.languages.join(', '));
```

Comment:
Just noting that our `AppLocalizationProvider` does something similar, we [pass in `nagivator.languages`](https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/index.tsx#L58) and [call `parseAcceptLanguage` here](https://github.com/mozilla/fxa/blob/main/packages/fxa-react/lib/AppLocalizationProvider.tsx#L169). 👍

Reply:
I got it from there. :)